onnxruntime/core/graph/graph.cc

```diff
@@ -2888,6 +2888,34 @@
   return ReplaceInitializedTensorImpl(std::move(new_initializer), false);
 }
 
+Status Graph::PopInitializedTensor(const std::string& initializer_name, ONNX_NAMESPACE::TensorProto& tensor_proto) {
```

Comment:
An expert in this graph implementation should also review.  Should this be layered under RemoveInitializedTensor?  The reason for differences is unclear.  Why does this not require the graph to be resolved, and why does it not support external data?